pkg/controller/containerjfr/containerjfr_controller_test.go

```diff
@@ -0,0 +1,563 @@
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	openshiftv1 "github.com/openshift/api/route/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/operator-framework/operator-sdk/pkg/log/zap"
+	rhjmcv1beta1 "github.com/rh-jmc-team/container-jfr-operator/pkg/apis/rhjmc/v1beta1"
+	"github.com/rh-jmc-team/container-jfr-operator/pkg/controller/containerjfr"
+	"github.com/rh-jmc-team/container-jfr-operator/pkg/controller/containerjfr/resource_definitions"
+	"github.com/rh-jmc-team/container-jfr-operator/test"
+)
+
+var _ = Describe("ContainerjfrController", func() {
+	var (
+		objs       []runtime.Object
+		client     client.Client
+		controller *containerjfr.ReconcileContainerJFR
+	)
+
+	JustBeforeEach(func() {
+		logf.SetLogger(zap.Logger())
+		s := test.NewTestScheme()
+
+		client = fake.NewFakeClientWithScheme(s, objs...)
+		controller = &containerjfr.ReconcileContainerJFR{
+			Client:        client,
+			Scheme:        s,
+			ReconcilerTLS: test.NewTestReconcilerTLS(client),
+		}
+	})
+
+	AfterEach(func() {
+		objs = nil
+	})
+
+	Describe("reconciling a request", func() {
+		Context("when reconciling", func() {
+			BeforeEach(func() {
+				objs = []runtime.Object{
+					test.NewContainerJFR(),
+				}
+			})
+			It("should create certificates", func() {
+				req := reconcile.Request{NamespacedName: types.NamespacedName{Name: "containerjfr", Namespace: "default"}}
+				result, err := controller.Reconcile(req)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(result).To(Equal(reconcile.Result{RequeueAfter: 5 * time.Second}))
+				certNames := []string{"containerjfr", "containerjfr-ca", "containerjfr-grafana"}
+				for _, certName := range certNames {
+					cert := &certv1.Certificate{}
+					err := client.Get(context.Background(), types.NamespacedName{Name: certName, Namespace: "default"}, cert)
+					Expect(err).ToNot(HaveOccurred())
+				}
+			})
+			It("should create routes", func() {
+				req := reconcile.Request{NamespacedName: types.NamespacedName{Name: "containerjfr", Namespace: "default"}}
+				result, err := controller.Reconcile(req)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(result).To(Equal(reconcile.Result{RequeueAfter: 5 * time.Second}))
+
+				// Update certificate status
+				MakeCertificatesReady(client)
+				InitializeSecrets(client)
+
+				// Check for routes, ingress configuration needs to be added as each
+				// one is created so that they all reconcile successfully
+				result, err = controller.Reconcile(req)
+				IngressConfig(client, *controller, req, false)
+			})
+		})
+		Context("succesfully creates required resources", func() {
```

Comment:
I would combine this Context with the previous one, using this description. Same thing for the minimal case.